osf/utils/workflows.py

```diff
@@ -94,6 +116,69 @@ def values(cls):
     }
 ]
 
+REGISTRATION_TRANSITIONS = [
+    {
+        'trigger': RegistrationTriggers.SUBMIT.value,
+        'source': [RegistrationStates.INITIAL.value],
+        'dest': RegistrationStates.PENDING.value,
+        'after': ['save_action', 'update_last_transitioned', 'submit_draft_registration', 'notify_submit'],
+    },
+    {
+        'trigger': RegistrationTriggers.ACCEPT.value,
+        'source': [RegistrationStates.PENDING.value, RegistrationStates.REJECTED.value],
+        'dest': RegistrationStates.ACCEPTED.value,
+        'after': ['save_action', 'update_last_transitioned', 'accept_draft_registration', 'notify_accept_reject'],
+    },
+    {
+        'trigger': RegistrationTriggers.REJECT.value,
+        'source': [RegistrationStates.PENDING.value],
+        'dest': RegistrationStates.REJECTED.value,
+        'after': ['save_action', 'update_last_transitioned', 'reject_draft_registration', 'notify_accept_reject'],
+    },
+    {
+        'trigger': RegistrationTriggers.FORCE_WITHDRAW.value,
+        'source': [RegistrationStates.PENDING.value, RegistrationStates.ACCEPTED.value],
+        'dest': RegistrationStates.REJECTED.value,
+        'after': ['save_action', 'update_last_transitioned', 'force_withdrawal', 'notify_accept_reject'],
+    },
+    {
+        'trigger': RegistrationTriggers.EMBARGO.value,
+        'source': [RegistrationStates.PENDING.value],
+        'dest': RegistrationStates.EMBARGO.value,
+        'after': ['save_action', 'update_last_transitioned', 'accept_draft_registration', 'embargo_registration', 'notify_embargo']
+    },
+    {
+        'trigger': RegistrationTriggers.REQUEST_EMBARGO_TERMINATION.value,
+        'source': [RegistrationStates.EMBARGO.value],
+        'dest': RegistrationStates.PENDING_EMBARGO_TERMINATION.value,
+        'after': ['save_action', 'update_last_transitioned', 'request_terminate_embargo', 'notify_embargo_termination']
+    },
+    {
+        'trigger': RegistrationTriggers.TERMINATE_EMBARGO.value,
+        'source': [RegistrationStates.PENDING_EMBARGO_TERMINATION.value],
+        'dest': RegistrationStates.ACCEPTED.value,
+        'after': ['save_action', 'update_last_transitioned', 'terminate_embargo', 'notify_embargo_termination']
+    },
+    {
+        'trigger': RegistrationTriggers.REQUEST_WITHDRAW.value,
```

Comment:
I'm not necessarily asking for a change here, but I do want to clarify. On the Preprints machine, there's a separate PreprintRequestMachine that is for handling the withdrawal requests. This causes [additional endpoints](https://www.notion.so/Registries-Moderation-Analysis-8b9a80f0fdb04da98a251112f4bf09be#43ee7060327241b88bbc6f4c93dfa560), because it's responding to a request rather than just moving around the machine states.

Now, front end has requested that we not do these additional endpoints, and instead just handle everything within out main state machine, which will make their lives easier. My question is, can we do everything we need to with just doing it this way, or are we going to be losing out on functionality somehow by not replicating this part of the preprint workflow.

Reply:
No functionality will be lost, virtually it's the same other the semantics.

Reply:
Awesome